src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import cx from "classnames";
+import React, { FC, ReactElement, useEffect, useState } from "react";
+import { SystemTheme, Theme, ThemeColor } from "./ThemeProviderConstants";
+import { generateThemeCssOverride } from "./ThemeProviderUtils";
+import { withStaticProps } from "../../types";
+
+export interface ThemeProviderProps {
+  theme?: Theme;
+  children: ReactElement;
+}
+
+const ThemeProvider: FC<ThemeProviderProps> & { systemThemes?: typeof SystemTheme; colors?: typeof ThemeColor } = ({
+  theme,
+  children
+}) => {
+  const [stylesLoaded, setStylesLoaded] = useState(false);
+
+  useEffect(() => {
+    if (!theme) {
+      return;
+    }
+
+    // Create a new style element
+    const styleElement = document.createElement("style");
+    styleElement.type = "text/css";
+    styleElement.id = theme.name;
```

Comment:
Name must be unique...., please enforce it

Reply:
done